src/BlobAdvanced.java

```diff
@@ -88,273 +95,258 @@ void runSamples() throws InvalidKeyException, URISyntaxException, IOException {
 
     /**
      * List containers sample.
-     * @param blobClient Azure Storage Blob Service
+     *
+     * @param blobServiceClient Azure Storage Blob Service
      */
-    private void listContainers(CloudBlobClient blobClient) throws URISyntaxException, StorageException {
+    private void listContainers(BlobServiceClient blobServiceClient) throws URISyntaxException, BlobStorageException {
         ArrayList<String> containerList = new ArrayList<>();
         try {
             System.out.println("Create containers");
             String prefix = UUID.randomUUID().toString();
             for (int i = 0; i < 5; i++) {
                 containerList.add(prefix + i);
-                blobClient.getContainerReference(prefix + i).create();
+                blobServiceClient.createBlobContainer(prefix + i);
             }
 
             System.out.println("List containers");
-            for (final CloudBlobContainer container : blobClient.listContainers(prefix)) {
-                System.out.printf("container name: %s%n", container.getName());
+            for (final BlobContainerItem blobContainerItem : blobServiceClient.listBlobContainers(new ListBlobContainersOptions().setPrefix(prefix), (Duration) null)) {
+                System.out.printf("container name: %s%n", blobContainerItem.getName());
             }
-        }
-        finally {
+        } finally {
             System.out.println("Delete containers");
             for (final String containerName : containerList) {
-                blobClient.getContainerReference(containerName).deleteIfExists();
+                blobServiceClient.getBlobContainerClient(containerName).delete();
```

Comment:
To translate this exactly, there should be a try/catch block that ignores the exception if it is a 404